nomad/datasource_nomad_job.go

```diff
@@ -0,0 +1,287 @@
+							Optional: true,
+						},
+						"promoted": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"desired_canaries": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"desired_total": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"placed_alloc": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"healthy_alloc": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"unhealthy_alloc": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"stable": &schema.Schema{
+				Description: "Job Stable",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"all_at_once": &schema.Schema{
+				Description: "Job All At Once",
+				Type:        schema.TypeBool,
+				Computed:    true,
+			},
+			"constraints": {
+				Description: "Job Constraints",
+				Computed:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"ltarget": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"rtarget": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"operand": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"update_strategy": {
+				Description: "Job Update Policy",
+				Computed:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"stagger": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"max_parallel": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+						"health_check": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"min_healthy_time": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"healthy_deadline": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"auto_revert": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"canary": {
+							Type:     schema.TypeInt,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"periodic_config": {
+				Description: "Job Periodic Configuration",
+				Computed:    true,
+				Type:        schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"enabled": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"spec": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"spec_type": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"prohibit_overlap": {
+							Type:     schema.TypeBool,
+							Optional: true,
+						},
+						"timezone": {
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+					},
+				},
+			},
+			"vault_token": {
+				Description: "Vault Token",
```

Comment:
If this is doing what I think it's doing--returning the vault token that the job is using--it should be marked `Sensitive`.